### PR TITLE
Propose /boot for encrypted root on PowerPC

### PR DESCRIPTION
**Pull request by k0da**

### CODE REVIEW

**Comment by aschnell** [comment]
As I already wrote in bugzilla this is a hack. For example if the proposal reuses partitions the ordering of the two /boot partitions generated in the proposal may change and then the sizes of /boot and PReP are also swapped. And a /boot partition with 7MiB to too small to even hold a filesystem.

**Comment by k0da** [comment]
aren't boot and boot2 different, boot will always be passed as first hence converted.

**Comment by aschnell** [comment]
You likely also missed the code in get_proposal_vm().

**Comment by aschnell** [comment]
It's not obvious that boot will always be passed as first and is always the one that has to be converted to PReP. Can you show that?

Even if, it's still a hack and we are required to improve code quality and not add one more hack. Things have changed in the last years.

**Comment by k0da** [comment]
We could improve SpecialBoothandling by looking at partition size, if it less than 2 cyls then convert it

**Comment by aschnell** [comment]
That is also a hack. The proper fix is to make the proposal code output the correct partitions from the beginning (as already stated in bugzilla).

File: testsuite/tests/empty-ppc64le4.rb [hunk 1]
```diff
@@ -0,0 +1,36 @@
+# encoding: utf-8
+
+# testedfiles: helper.rb
+
+module Yast
```
Comment (kobliha) [inline testsuite/tests/empty-ppc64le4.rb]: I'm missing a description of this test. Please add at least a comment what it tests and how. And what is the expected output.

**Comment by kobliha** [comment]
I agree with Arvin, that hacks must not be added to the code anymore. I'm expecting a cleanup rather than adding something that will take five times more to fix later. It might help to have a chat with Arvin/HuHa at first, before changing the code. It might be faster.

File: testsuite/tests/empty-ppc64le4.rb (hunk 1 above)
Comment (k0da) [inline testsuite/tests/empty-ppc64le4.rb]: I don't see any other tests are doing this.

**Comment by k0da** [comment]
@kobliha do you want refactor at RC1?

File: testsuite/tests/empty-ppc64le4.rb (hunk 1 above)
Comment (kobliha) [inline testsuite/tests/empty-ppc64le4.rb]: Yes, you are right, I would like to see it for all the tests from now on. Everything is for the first time once...

**Comment by kobliha** [comment]
@k0da No, what I want is a working code that does not add more hacks. I know that you want to help us to "make it work", but please, consult with @aschnell and @shundhammer how to make it properly. As I've already written in Bugzilla, we need a specification for location and appearance of the PReP partition before implementing anything. We can't be sure that it will work otherwise.

File: src/modules/StorageProposal.rb [hunk 2]
```diff
@@ -5624,11 +5624,16 @@ def get_inst_prop_vm(target, key)
       Ops.set(ret, "target", target)
 
       boot2 = {}
-      if GetProposalEncrypt() && Partitions.EfiBoot
+      if GetProposalEncrypt() && (Partitions.EfiBoot||Partitions.PrepBoot)
```
Comment (kobliha) [inline src/modules/StorageProposal.rb]: BTW, and this is pure nitpicking, Ruby style requires spaces around `||`

**Comment by teclator** [comment]
@k0da do you plan to do something with this PR, it is currently outdated and the related bug is already closed so probably we could also close it.

**Comment by jreidinger** [comment]
storage is replaced by storage-ng so closing.

**Pull request closed without merging**